src/Surfnet/ServiceProviderDashboard/Infrastructure/Manage/Client/QueryClient.php

```diff
@@ -117,9 +122,17 @@ public function getMetadataXmlByManageId($manageId)
     public function findByManageId($manageId)
     {
         try {
+            // TODO: investigate if we can add the protocol to the param list of this method to prevent the try/retry
```

Comment:
> I presume this was a notice for yourself during development?

It was initially, but it was not a triviality to add the additional parameter, as this method is used in many different contexts. Making it hard to always include the protocol. I left this todo for future reference and to pick your brain during this review. 

What do you think about this less than elegant solution? Is it acceptable for now or should we invest in a better, more time-consuming solution?

Reply:
I'm curious what the response from Okke was? For now I think this will be sufficient but if it could be changed with minimal effort then I'll go for that. It's good that this is contained in the infrastructure layer.

Reply:
Due to the way the nosql backend of manage is set up, it is not easy to query over the two protocols. For now that is out of scope. The best thing he could do was to add the protocol to every response from the search API.

We both had a good laugh about the current solution :) And I'm now working on fixing this.